src/prpy/util.py

```diff
@@ -751,6 +751,88 @@ def IsAtTrajectoryStart(robot, trajectory):
     # If all joints match, return True.
     return True
 
+def IsAtTrajectoryEnd(robot, trajectory):
+    """
+    Check if robot's DOFs match the end configuration of a trajectory.
+
+    This function examines the current DOF values of the specified robot and
+    compares these values to the last waypoint of the specified trajectory.
+    If every DOF value specified in the trajectory differs by less than the
+    DOF resolution of the specified joint/axis then it will return True.
+    Otherwise, it returns False.
+
+    @param robot: the robot whose active DOFs will be checked
+    @param trajectory: the trajectory whose end configuration will be checked
+    @returns: True if the robot's active DOFs match the given trajectory
+              False if one or more active DOFs differ by DOF resolution
+    """
+    cspec = trajectory.GetConfigurationSpecification()
+    needs_base = HasAffineDOFs(cspec)
+    needs_joints = HasJointDOFs(cspec)
+    
+    if needs_base and needs_joints:
+        raise ValueError('Trajectories with affine and joint DOFs are not supported in IsAtTrajectoryEnd()')
+
+    if trajectory.GetEnv() != robot.GetEnv():
+        raise ValueError('The environment attached to the trajectory does not match the environment attached to the robot in IsAtTrajectoryEnd()')
+
+    if needs_base:
+        raise ValueError('Trajectories with affine DOFs are not supported in IsAtTrajectoryEnd()')
+
+    else:
+        last_waypoint_idx = trajectory.GetNumWaypoints() - 1
```

Comment:
This will cause an `IndexError` on an empty trajectory. I see two options for how to handle this:

1. check for an empty trajectory and throw an error with a good error message
2. return `True`

@psigen Any preference?